src/DynamoCore/Core/NodeModelAssemblyLoader.cs

```diff
@@ -198,6 +199,22 @@ internal static bool ContainsNodeModelSubType(Assembly assem)
         {
             return assem.GetTypes().Any(IsNodeSubType);
         }
+        internal static bool ContainsNodeViewCustomizationType(Assembly assem)
+        {
+            return GetCustomizationTypesUsingReflection(assem).Any();
+        }
+
+        internal static IEnumerable<Type> GetCustomizationTypesUsingReflection(Assembly assem)
+        {
+
+            var customizerType = Type.GetType("Dynamo.Wpf.INodeViewCustomization`1,DynamoCoreWpf");
```

Comment:
shouldn't we use the fully qualified name here ?
Any chance of ambiguity when running in headless mode (no wpf ) ? Or we always load the wpf assemblies

Reply:
Hey @pinzart90 - what else would be included in the fully qualified name? this is `namespace.type ,assembly`

In headless mode this type will not be loaded, but theres no reason to load or apply view customizations then anyway since WPF types will not be loaded either.

Reply:
let me give it some thought, but would you agree with the following points:
1. If dynamo is running headless on a system where `DynamoCoreWpf` can actually be loaded, then loading it should not cause issues - because the DynamoView will still never be instantiated even if the assembly is loaded, and so customizations will not be applied (this occurs when node views are created)

2. If dynamo is running headless on a system where `DynamoCoreWpf` cannot be loaded, then this call should return null and we won't try to load any view customizations which we couldn't apply anyway.

I do like your suggestions, though they may be overkill - hard to tell I guess. 

Checking if the assembly has already been loaded may work (I'll have to see if that works based on when this function gets called... it's normally before the view is created...)
ReflectionOnlyLoad context may be the way to go since we only care about checking for the existence of types.

Reply:
1. I guess the only issue will be that we're loading DynamoCoreWpf + all direct dependencies when there is no need to.
2. It might throw an exception...we should double check the docs.

ReflectionOnlyLoad will not load any of the dependencies (afaik). So trying to get a specific type (or a list of all types) might fail.

If we do introduce a breaking change in the interface and an older dynamocorewpf assembly is loaded, I guess the TypeExtensions.ImplementsGeneric(customizerType, t) will simply fail....
Should not be a common occurrence though. Not sure how we could guard against such a situation without development difficult (as you mentioned)...

Reply:
If `Checking if the assembly has already been loaded` does not seem reliable enough, then I am fine with the current state of things